ui/components/src/Legend/ListLegendItem.tsx

```diff
@@ -11,20 +11,59 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import React from 'react';
+import React, { forwardRef, useState, useEffect } from 'react';
 import { Box, ListItemText, ListItem, ListItemProps } from '@mui/material';
 import { LegendItem } from '../model';
 import { combineSx } from '../utils';
 import { LegendColorBadge } from './LegendColorBadge';
 
-interface ListLegendItemProps extends ListItemProps {
+interface ListLegendItemProps extends ListItemProps<'div'> {
   item: LegendItem;
+
+  /**
+   * When `true`, will keep labels to a single line with overflow ellipsized. The
+   * full content will be shown on hover.
+   *
+   * When `false` or unset, will show the full label.
+   */
+  truncateLabel?: boolean;
+
+  /**
+   * Called when the layout of the legend item changes as a result of the hover
+   * behavior when `truncateLabel` is `true`.
+   */
+  onLayoutChange?: () => void;
```

Comment:
Need these props to get the ellipsized truncation + the virtualization working properly together.